source/initial_temperature/random_gaussian_perturbation.cc

```diff
@@ -0,0 +1,208 @@
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::Chunk<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::Chunk<dim> &chunk_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::Chunk<dim>> (this->get_geometry_model());
+
+          for (unsigned int d=0; d<dim; ++d)
+            {
+              min_coordinates[d] = -chunk_geometry_model.outer_radius();
+              max_coordinates[d] = chunk_geometry_model.outer_radius();
+            }
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::Box<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::Box<dim> &box_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::Box<dim>> (this->get_geometry_model());
+
+          min_coordinates = box_geometry_model.get_origin();
+          max_coordinates = min_coordinates + box_geometry_model.get_extents();
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::TwoMergedBoxes<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::TwoMergedBoxes<dim> &two_merged_boxes_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::TwoMergedBoxes<dim>> (this->get_geometry_model());
+
+          min_coordinates = two_merged_boxes_geometry_model.get_origin();
+          max_coordinates = min_coordinates + two_merged_boxes_geometry_model.get_extents();
+        }
+      else
+        AssertThrow (false,
+                     ExcMessage ("Not a valid geometry model for the initial conditions model "
+                                 "random Gaussian perturbation."));
+
+      // Initialize random locations of perturbations and check they are in the model domain.
+      // Use a fixed number as seed for random generator.
+      // This is important if we run the code on more than 1 processor.
+      std::srand(1);
+
+      for (unsigned int n=0; n<n_perturbations; ++n)
+        {
+          do
+            {
+              for (unsigned int d=0; d<dim; ++d)
+                perturbation_centers[n][d] = min_coordinates[d] + (double)(std::rand() % int(max_coordinates[d] - min_coordinates[d]));
+            }
+          while (!this->get_geometry_model().point_is_in_domain(perturbation_centers[n]));
+
+          const int sign = 2 * (std::rand() % 2) - 1;
+          perturbation_magnitudes[n] = sign * (double)(std::rand() % int(max_magnitude));
+        }
+    }
+
+
+
+    template <int dim>
+    double
+    RandomGaussianPerturbation<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      double temperature_perturbation = 0;
+      for (unsigned int n=0; n<n_perturbations; ++n)
+        {
+          double exponent = 0;
+          for (unsigned int d=0; d<dim; ++d)
+            exponent += (position[d] - perturbation_centers[n](d)) * (position[d] - perturbation_centers[n](d))
+                        / (2.*width*width);
```

Comment:
Isnt this the same as (maybe simpler to read):
```suggestion
          const double distance_square = position.distance_square(perturbation_centers[n]);
          const double exponent = distance_square / (2.*width*width);
```